modules/ding_gatewayf/modules/ding_gatewayf_registration/ding_gatewayf_registration.module

```diff
@@ -0,0 +1,456 @@
+<?php
+/**
+ * @file
+ * Handle user self registration after login into gatewayf.
+ */
+
+define('DING_GATEWAYF_REGISTRATION_ACCEPTANCE_URL', 'gatewayf/registration/acceptance');
+define('DING_GATEWAYF_REGISTRATION_INFORMATION_URL', 'gatewayf/registration/information');
+define('DING_GATEWAYF_REGISTRATION_SUCCESS_URL', 'gatewayf/registration/success');
+
+/**
+ * Implements hook_menu().
+ */
+function ding_gatewayf_registration_menu() {
+  $items = array();
+
+  $items[DING_GATEWAYF_REGISTRATION_INFORMATION_URL] = array(
+    'page callback' => 'ding_gatewayf_registration_information',
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  );
+
+  $items[DING_GATEWAYF_REGISTRATION_ACCEPTANCE_URL] = array(
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_gatewayf_registration_acceptance_form'),
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  );
+
+  $items[DING_GATEWAYF_REGISTRATION_SUCCESS_URL] = array(
+    'page callback' => 'ding_gatewayf_registration_success',
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function ding_gatewayf_registration_theme($existing, $type, $theme, $path) {
+  return array(
+    'ding_gatewayf_registration_link' => array(
+      'template' => 'ding-gatewayf-registration-link',
+      'variables' => array(
+        'link' => NULL,
+      ),
+      'path' => $path . '/templates',
+    ),
+    'ding_gatewayf_registration_information' => array(
+      'template' => 'ding-gatewayf-registration-information',
+      'variables' => array(
+        'title' => array(),
+        'content' => array(),
+        'link' => NULL,
+      ),
+      'path' => $path . '/templates',
+    ),
+    'ding_gatewayf_registration_success_page' => array(
+      'template' => 'ding-gatewayf-registration-success',
+      'variables' => array(
+        'title' => array(),
+        'content' => array(),
+      ),
+      'path' => $path . '/templates',
+    ),
+  );
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function ding_gatewayf_registration_block_info() {
+  return array(
+    'registration' => array(
+      'info' => t('Ding GateWAYF self registration link'),
```

Comment:
I have simply taken the fast way and provided blocks the same way as the previous wayf module ding_wayf do.

If this should be re-done it would require another ticket.